### PR TITLE
Add a cronjob to gc PipelineRuns

### PR DESCRIPTION
**Pull request by LinuxSuRen**
Upstream is https://github.com/kubesphere/ks-devops/pull/237

### CODE REVIEW

File: charts/ks-devops/values.yaml [hunk 1]
```diff
@@ -14,6 +14,20 @@ image:
     tag: master
   pullPolicy: IfNotPresent
 
+gcJobs:
+  # gcJobs.maxAge -- Max age from which `PipelineRun`s will be deleted
+  maxAge: 168h
+  # gcJobs.maxCount -- Max count from which `PipelineRun`'s will be deleted
```
Comment (JohnNiang) [inline charts/ks-devops/values.yaml:20]: `PipelineRun`'s -> `PipelineRun`s

File: charts/ks-devops/templates/cronjob-gc.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ include "ks-devops.apiserver-fullname" . }}
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cronjob-gc.yaml:4]: Why we use full name of apiserver here?

File: charts/ks-devops/templates/cronjob-gc.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  labels:
+    app: devops-pipelinerun-gcjobs
+spec:
+  concurrencyPolicy: {{ .Values.gcJobs.concurrencyPolicy }}
+  failedJobsHistoryLimit: {{ .Values.gcJobs.failedJobsHistoryLimit }}
+  jobTemplate:
+    metadata:
+      creationTimestamp: null
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cronjob-gc.yaml:12]: I think we don't need creationTimestamp field here. Please see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/.

File: charts/ks-devops/templates/cronjob-gc.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    spec:
+      template:
+        metadata:
+          creationTimestamp: null
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cronjob-gc.yaml:16]: Same as above

File: charts/ks-devops/templates/cronjob-gc.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          labels:
+            app: {{ include "ks-devops.apiserver-fullname" . }}
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cronjob-gc.yaml:18]: Same as above.

File: charts/ks-devops/values.yaml (hunk 1 above)
Comment (LinuxSuRen) [reply charts/ks-devops/values.yaml:20]: Why?

Comment (JohnNiang) [reply charts/ks-devops/values.yaml:20]: Please see line#18.

Comment (LinuxSuRen) [reply charts/ks-devops/values.yaml:20]: I see

File: charts/ks-devops/templates/cronjob-gc.yaml (hunk 2 above)
Comment (LinuxSuRen) [reply charts/ks-devops/templates/cronjob-gc.yaml:4]: How about changing it to `ks-devops.fullname`

**Comment by JohnNiang** [comment]
LGTM

**Pull request merged**